File: app/src/main/java/space/pal/sig/model/dto/LaunchDto.java
package space.pal.sig.model.dto;

import com.google.gson.annotations.SerializedName;

import java.util.List;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class LaunchDto {

    @SerializedName("name")
    private String name;
    @SerializedName("windowstart")
    private String windowstart;
    @SerializedName("windowend")
    private String windowend;
    @SerializedName("net")
    private String net;
    @SerializedName("wsstamp")
    private Long wsstamp;
    @SerializedName("westamp")
    private Long westamp;
    @SerializedName("netstamp")
    private Long netstamp;
    @SerializedName("infoURLs")
    private String[] infoURLs;
    @SerializedName("vidURLs")
    private String[] vidURLs;
    @SerializedName("status")
    private Integer status;
    @SerializedName("holdreason")
    private String holdreason;
    @SerializedName("failreason")
    private String failreason;
    @SerializedName("hashtag")
    private String hashtag;
    @SerializedName("location")
    private LocationDto location;
    @SerializedName("rocket")
    private RocketDto rocket;
    @SerializedName("missions")
    private List<MissionDto> missions;
    @SerializedName("lsp")
    private SpaceProviderDto lsp;
}

File: app/src/main/java/space/pal/sig/model/dto/LaunchDto.kt
package space.pal.sig.model.dto

import com.google.gson.annotations.SerializedName
import space.pal.sig.model.entity.Launch

/**
SpaceNow
Created by Catalin on 12/6/2020
 **/
data class LaunchDto(
        @SerializedName("id")
        val id: String,
        @SerializedName("flight_number")
        val flightNumber: Long,
        @SerializedName("name")
        val name: String,
        @SerializedName("date_utc")
        val dateUtc: String,
        @SerializedName("date_unix")
        val dateUnix: Long,
        @SerializedName("date_local")
        val dateLocal: String,
        @SerializedName("static_fire_date_utc")
        val staticFireDateUtc: String?,
        @SerializedName("static_fire_date_unix")
        val staticFireDateUnix: Long?,
        @SerializedName("tbd")
        val tbd: Boolean,
        @SerializedName("net")
        val net: Boolean,
        @SerializedName("window")
        val window: Long?,
        @SerializedName("rocket")
        val rocket: String,
        @SerializedName("success")
        val success: Boolean?,
        @SerializedName("failures")
        val failures: List<Failure>,
        @SerializedName("upcoming")
        val upcoming: Boolean,
        @SerializedName("details")
        val details: String?,
        @SerializedName("crew")
        val crew: List<String>,
        @SerializedName("ships")
        val ships: List<String>,
        @SerializedName("capsules")
        val capsules: List<String>,
        @SerializedName("payloads")
        val payloads: List<String>,
        @SerializedName("launchpad")
        val launchpad: String?,
        @SerializedName("links")
        val links: DataLinks,
        @SerializedName("auto_update")
        val autoUpdate: Boolean,
        @SerializedName("cores")
        val cores: List<LaunchCore>,
        @SerializedName("fairings")
        val fairings: LaunchFairing?
) {
    fun toLaunch(): Launch {
        return Launch(
                id, flightNumber, name, dateUtc, dateUnix * 1000,
                dateLocal, staticFireDateUtc, staticFireDateUnix,
                tbd, net, window, rocket, success, failures,
                upcoming, details, crew, ships, capsules, payloads,
                launchpad, links, autoUpdate, cores, fairings
        )
    }
}
